### ISSUE TITLE
`default_entity_manager` config param is not inherited from parent config file

### ISSUE THREAD

**Issue by bravik** [doctrine/DoctrineBundle#1337]
This line in Configuratation:
https://github.com/doctrine/DoctrineBundle/blob/09339f7df6fc620ce805f52ca0c7424543d30515/DependencyInjection/Configuration.php#L376

makes `default_entity_manager` (=<some_none_default_value>) param defined in `config/packages/doctrine.yml`
been overriden by `config/packages/prod/doctrine.yml` without this option.
Instead of the value from `config/packages/doctirne.yml` we have `default` value.
Which is conter-intuitive and took a couple of time to investigate mapping issues.

Example:
```yaml
#c onfig/packages/doctrine.yml
doctrine:
  # ...
    orm:
        auto_generate_proxy_classes: true
        default_entity_manager: app
        entity_managers:
            app:
               # ...
            module:
               #...
```

```yaml
#config/packages/prod/doctrine.yml
# The default one, generated by flex

doctrine:
    orm:
        auto_generate_proxy_classes: false
        metadata_cache_driver:
            type: pool
            pool: doctrine.system_cache_pool
        query_cache_driver:
            type: pool
            pool: doctrine.system_cache_pool
        result_cache_driver:
            type: pool
            pool: doctrine.result_cache_pool
```

Expected: `app`
Given: `default`

**Comment by ostrolucky** [comment]
Can be also tested via `bin/console debug:container --parameter=doctrine.default_entity_manager --env=prod` and `bin/console debug:container --parameter=doctrine.default_entity_manager --env=dev`, respectively. 

If anyone want to work on this, I think solution would be to extract this normalization https://github.com/doctrine/DoctrineBundle/blob/d6b3c37804539a24ba8a7d647a6144cab2f13242/DependencyInjection/Configuration.php#L376-L377 to DoctrineExtension

**Comment by ostrolucky** [comment]
Patch has been merged into 2.4. It should prevent similar issues in future
